### PR TITLE
update monitor csigpki verification

### PR DESCRIPTION
**Pull request by g-k**
Changes:

* ci: run monitor unit tests

This causes: 

```
2021/02/01 21:50:05 Chain verification failed with: Certificate 0 "normandy.content-signature.mozilla.org" expires in less than 15 days: notAfter=2017-11-07 14:02:37 +0000 UTC
--- FAIL: TestVerifyWronglyOrderedChain (0.00s)
    contentsignaturepki_test.go:64: Failed to retrieved certificate chain: failed to verify certificate chain: x509: certificate has expired or is not yet valid: current time 2021-02-01T21:50:05Z is after 2021-01-31T22:06:45Z
FAIL
```

https://app.circleci.com/pipelines/github/mozilla-services/autograph/321/workflows/9aee1cbc-1636-4b14-8639-552a78a2d9cc/jobs/5335/parallel-runs/0/steps/0-108

* tools: update expired CS intermediate in WronglyOrderedChain
* signer: verify PEM decoding in csigpki GetX5U

golang's encoding/pem.Decode returns nil when no PEM data is found https://golang.org/pkg/encoding/pem/#Decode

* tools: verify csign with monitor instead of app signer

see https://github.com/mozilla-services/autograph/issues/593 for the difference between the implementations

### CODE REVIEW

File: tools/autograph-monitor/monitor.go [hunk 1]
```diff
@@ -153,7 +153,7 @@ func monitor() (err error) {
 			err = verifyContentSignature(response)
 		case contentsignaturepki.Type:
 			log.Printf("Verifying content signature pki from signer %q", response.SignerID)
-			err = contentsignaturepki.Verify(response.X5U, response.Signature, []byte(inputdata))
+			err = verifyContentSignature(response)
```
Comment (ajvb) [inline tools/autograph-monitor/monitor.go:156]: Apologies, as I'm new to reviewing autograph code, but what is the implication of this change?

**Review by ajvb — APPROVED** [review]
LGTM!

File: tools/autograph-monitor/monitor.go (hunk 1 above)
Comment (g-k) [reply tools/autograph-monitor/monitor.go:156]: Thanks for taking a look AJ!

The `contentsignaturepki.Verify` from `signer/contentsignaturepki` https://github.com/mozilla-services/autograph/blob/master/signer/contentsignaturepki/contentsignature.go#L321 just verifies the leaf or EE in the root -> intermediate -> EE chain.

The `verifyContentSignature` from `tools/autograph-monitor` https://github.com/mozilla-services/autograph/blob/d58a6b57fb2e7f6f72cd29496e0d25080b040a70/tools/autograph-monitor/contentsignaturepki.go#L33 verifies the response data and uses [verifyCertChain](https://github.com/mozilla-services/autograph/blob/d58a6b57fb2e7f6f72cd29496e0d25080b040a70/tools/autograph-monitor/contentsignaturepki.go#L88) to check the full root -> intermediate -> EE chain. It sends a soft SNS notification if any cert has <30 days until it expires and errors if any cert is <15 days to expiration or the cert isn't currently valid (i.e. cert NotBefore < now).

There's also a third chain verification implementation in the code that uploads the chain to S3, I filed https://github.com/mozilla-services/autograph/issues/593 to clean them up.

Comment (ajvb) [reply tools/autograph-monitor/monitor.go:156]: Ok awesome! Thank you for the explanation, this sounds great to me.

**Comment by g-k** [comment]
Thanks AJ! I hope you don't mind me running some autograph reviews by you. It's good to get a sanity check.

**Comment by ajvb** [comment]
@g-k I don't mind at all!

**Pull request merged**